src/modules/sync/diff/transformToAddModel.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  ContentTypeElements,
+  ContentTypeModels,
+  ContentTypeSnippetModels,
+  TaxonomyModels,
+} from "@kontent-ai/management-sdk";
+
+import {
+  assetAttributeName,
+  assetExternalIdAttributeName,
+  itemExternalIdLinkAttributeName,
+  itemLinkAttributeName,
+} from "../../../constants/richText.js";
+import { RequiredCodename } from "../../../utils/types.js";
+import { customAssetCodenameAttributeName, customItemLinkCodenameAttributeName } from "../../constants/syncRichText.js";
+import { ContentTypeSnippetsSyncModel, ContentTypeSyncModel, TaxonomySyncModel } from "../types/fileContentModel.js";
+import { SyncGuidelinesElement } from "../types/syncModel.js";
+
+export const transformTaxonomyToAddModel = (
+  taxonomy: TaxonomySyncModel,
+): RequiredCodename<TaxonomyModels.IAddTaxonomyRequestModel> => ({
+  ...taxonomy,
+  terms: taxonomy.terms.map(transformTaxonomyToAddModel),
+});
+
+export const transformTypeToAddModel =
+  (params: TransformElementParams) =>
+  (type: ContentTypeSyncModel): RequiredCodename<ContentTypeModels.IAddContentTypeData> => ({
+    ...type,
+    content_groups: type.content_groups?.slice(),
+    elements: type.elements.map(transformElementToAddModel(params)) as ContentTypeElements.Element[],
+  });
+
+export const transformSnippetToAddModel =
+  (params: TransformElementParams) =>
+  (snippet: ContentTypeSnippetsSyncModel): RequiredCodename<ContentTypeSnippetModels.IAddContentTypeSnippetData> => ({
+    ...snippet,
+    elements: snippet.elements.map(transformElementToAddModel(params)) as ContentTypeElements.Element[],
+  });
+
+type TransformElementParams = Readonly<{
+  targetItemsReferencedFromSourceByCodenames: ReadonlyMap<string, Readonly<{ id: string; codename: string }>>;
+  targetAssetsReferencedFromSourceByCodenames: ReadonlyMap<string, Readonly<{ id: string; codename: string }>>;
+}>;
+
+const transformElementToAddModel = (params: TransformElementParams) =>
+(
+  el: ContentTypeSyncModel["elements"][number],
+): ContentTypeElements.Element | { type: "subpages"; default?: ContentTypeElements.ILinkedItemsElement["default"] } => { // to fix SDK types bug
+  switch (el.type) {
+    case "number":
+    case "snippet":
+    case "url_slug":
+    case "date_time":
+    case "text":
+      return el;
+    case "asset":
+      return {
+        ...el,
+        default: el.default ? { global: { value: [...el.default.global.value] } } : undefined,
+      };
+    case "custom":
+      return { ...el, allowed_elements: el.allowed_elements?.slice() };
```

Comment:
Yeah, it is. Another option is to cast it, but it has to go through unknown. I want to create an issue on the SDK to make all the arrays readonly to avoid this stupidity.

Reply:
I changed it to cast the arrays instead of recreating them, but the SDK types still need to change to get rid of the code anyway.